docs/src/languages/java/migration.md

```diff
@@ -0,0 +1,172 @@
+
+## Minimum Requirement
+
+To run a Java application at Platform.sh you will need: 
+
+* [A supported Java version]({{< relref "/languages/java.md#supported-versions" >}})
+* [A build management tool]({{< relref "/languages/java.md#support-build-automation" >}})
+  * [Gradle](https://docs.gradle.org/current/userguide/gradle_wrapper.html)
+  * [Maven](https://maven.apache.org/) 
+  * [Maven Wrapper](https://www.baeldung.com/maven-wrapper)
+  * [Ant](https://ant.apache.org/)
+* A Git Repository:
+  * [GitHub]({{< relref "integrations/source/github.md" >}})
+  * [BitBucket]({{< relref "integrations/source/bitbucket.md" >}})
+  * [GitLab]({{< relref "integrations/source/gitlab.md" >}})
+  * The default Git repository provided by Platform.sh
+
+{{< note >}}
+A container application cannot be bigger than **8 GB** of memory, see  [tunning](https://docs.platform.sh/languages/java/tuning.html) to more details.
+{{< /note >}}
+
+## Monolith/Single Application
+
+To start a Java application, you need to understand the [Platform.sh structure](https://docs.platform.sh/overview/structure.html).  At minimum you will need three [YAML files](https://docs.platform.sh/configuration/yaml.html): 
+
+1. [Application]({{< relref "configuration/app.md" >}})
+2. [Route]({{< relref "configuration/routes.md" >}})
+
+### Application
+
+```yaml
+# .platform.app.yaml
+
+name: app
+type: "java:<version>" 1
+disk: 1024
+hooks:
+    build: 2
+web:
+    commands:
+        start: 3
+```
+
+1. [A Java version]({{< relref "languages/java.md#supported-versions" >}}), e,g.: `java:11`
+2. [The build defines what happens when building the application]({{< relref "configuration/app/build.md#build" >}}), This build process will typically generate an executable file such as a uber-jar e.g.: `mvn clean package`
+3. [The commands key defines the command to launch the application]({{< relref "configuration/app/web.md#commands" >}}). E.g.:  `java -jar file.jar`
+4. In the start's command needs to receive the port where the application will execute thought the `PORT` environment. That is trivial if your application follows the port bind principle. E.g.: `java -jar jar --port=$PORT`
+
+### Route
+
+```yaml
+# .platform/routes.yaml
+
+"https://{default}/":
+  type: upstream
+  upstream: 1
+"https://www.{default}/":
+  type: redirect
+  to: "https://{default}/"
+```
+
+1. It defines the application will link in the route, e.g.: `"app:http"`
+
+{{< note >}}
+
+Be aware that the container has a maximum memory if the application is stateless to increase a more powerful computer, there is the strategy to use a [load balancer](https://community.platform.sh/t/how-to-configure-load-balancer-in-a-single-application/553).
+
+Be aware that after the build, it creates a read-only system. You have the [mount option to create a writable folder]({{< relref "configuration/app/storage.md#mounts" >}}).
```

Comment:
This line does not belong in the routes section.  It should be in the application section.